src/Sulu/Bundle/WebsiteBundle/DataCollector/SuluCollector.php

```diff
@@ -76,39 +81,52 @@ public function collect(Request $request, Response $response, ?\Throwable $excep
         $this->data['resource_locator_prefix'] = $requestAttributes->getAttribute('resourceLocatorPrefix');
 
         $structure = null;
-        if ($request->attributes->has('_route_params')) {
-            $params = $request->attributes->get('_route_params');
-            if (isset($params['structure'])) {
-                /** @var StructureInterface $structureObject */
-                $structureObject = $params['structure'];
-
-                $structure = [
-                    'id' => $structureObject->getUuid(),
-                    'objectClass' => $structureObject::class,
-                    'path' => $structureObject->getPath(),
-                    'nodeType' => $structureObject->getNodeType(),
-                    'internal' => $structureObject->getInternal(),
-                    'nodeState' => $structureObject->getNodeState(),
-                    'published' => $structureObject->getPublished(),
-                    'publishedState' => $structureObject->getPublishedState(),
-                    'navContexts' => $structureObject->getNavContexts(),
-                    'shadowLocales' => $structureObject->getShadowLocales(),
-                    'contentLocales' => $structureObject->getContentLocales(),
-                    'shadowOn' => $structureObject->getIsShadow(),
-                    'shadowBaseLanguage' => $structureObject->getShadowBaseLanguage(),
-                    'template' => $structureObject->getKey(),
-                    'originTemplate' => $structureObject->getOriginTemplate(),
-                    'hasSub' => $structureObject->getHasChildren(),
-                    'creator' => $structureObject->getCreator(),
-                    'changer' => $structureObject->getChanger(),
-                    'created' => $structureObject->getCreated(),
-                    'changed' => $structureObject->getChanged(),
-                ];
+        if ($request->attributes->has('object')) {
+            $object = $request->attributes->get('object');
+
+            $structure = [];
+            if ($object instanceof DimensionContentInterface) {
+                $parent = $object->getResource();
+
+                if (\method_exists($parent, 'getUuid')) {
```

Comment:
the resource must implement the `ContentRichEntityInterface` therefore it must have a `getId()´method. In the long run it should be replaced with `getUuid` but we should use the existing interface method here